src/codegen/function-declaration.ts

```diff
@@ -31,6 +31,13 @@ export default class CodeGenFuncDecl {
         });
       }
     });
+
+    if (node.type && common.findRealType(funcReturnType).isStructTy()) {
+      const typeLiteral = node.type! as ts.TypeLiteralNode;
+      const fields = common.buildStructMaps(funcReturnType as llvm.StructType, typeLiteral);
+
+      this.cgen.symtab.set(func.name, { inner: func, deref: 0, fields });
```

Comment:
I'm not understand why a `Function` should have `fields`, It is illogical.

I think the a Function has a return type, and the type contain `fields`.

Reply:
1. return {a: 10, b:  “str”}
2. return {c: 10, d:  “str”}

They have the same type, but they don't have the same name, right?

Because of the object text, I cannot create the same 'fields' for a type.

Reply:
What if think of them as two types?

Reply:
So I'm going to create a lot of extra types, which I don't think is necessary.

Reply:
In ts, there are two types, see

```ts
let a = {name: 'ss'}
let b = {eman: 'ss'}

function echo(i: {name: string}) {
    console.log(a)
}

echo(a)
echo(b)  // Argument of type '{ eman: string; }' is not assignable to parameter of type '{ name: string; }'.
// Property 'name' is missing in type '{ eman: string; }' but required in type '{ name: string; }'```

Reply:
````ts
let a = { name: 'ss', d: 10 };
let b = { name: 'ss', c: 10, d: 10, f: 'str' };

function echo(i: { name: string; d: number }) {
  console.log(i);
}

echo(a); // { name: 'ss', d: 10 }
echo(b); // { name: 'ss', c: 10, d: 10, f: 'str' }
````

Reply:
So, we can create an anonymous type and restrict the fields strictly, but lose some of the features of `ts`.

Reply:
I don't think they are one type, even in your example code. It's very clearly if I update your codes:

```
let a = { name: 'ss', d: 10 };
let b = { name: 'ss', c: 10, d: 10, f: 'str' };

function echo(i: { name: string, c: number, d: number, f: string }) {
  console.log(i);
}

echo(a); // x
echo(b);
```

Reply:
And can you give a reasonable explanation: Why a function and a variable has `fields`,  not a type has `fields`?

Reply:
You can think of `ObjectLiteral` as duck types, like interfaces in `golang`, where any object that satisfies the declaration can be passed in as a parameter.

Reply:
But,  it's all up to you.

Reply:
It's not about listen to me or listen to you, It's about our program.

I studied the `ts-llvm` 's source code, It get the index of field name from `ts.Type` by  `ts.TypeChecker`, maybe we should also add `ts.TypeChecker`, It seems makes the code very simple.

See: https://github.com/ts-llvm/ts-llvm/blob/master/src/codegen/expression.ts#L282